src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -419,7 +470,41 @@ private async Task<bool> NewerManifestPackageExists(ManifestId manifest)
             }
         }
 
-        private IEnumerable<(ManifestId Id, ManifestVersionWithBand ManifestWithBand)> ParseRollbackDefinitionFile(string rollbackDefinitionFilePath)
+        public IEnumerable<ManifestVersionUpdate> ParseRollbackDefinitionFiles(IEnumerable<string> rollbackFilePaths)
+        {
+            if (rollbackFilePaths.Count() == 1)
+            {
+                return CalculateManifestRollbacks(rollbackFilePaths.Single());
+            }
+
+            var currentManifestIds = GetInstalledManifestIds();
+            // Create a single workload set that includes all the others
+            List<(ManifestId, ManifestVersionWithBand)> fullSet = new();
+            foreach (var rollbackFile in rollbackFilePaths)
+            {
+                var rollbacks = ParseRollbackDefinitionFile(rollbackFile, _sdkFeatureBand);
+
+                var unrecognizedManifestIds = rollbacks.Where(rollbackManifest => !currentManifestIds.Contains(rollbackManifest.Id));
+                if (unrecognizedManifestIds.Any())
+                {
+                    _reporter.WriteLine(string.Format(LocalizableStrings.RollbackDefinitionContainsExtraneousManifestIds, rollbackFile, string.Join(" ", unrecognizedManifestIds)).Yellow());
+                    rollbacks = rollbacks.Where(rollbackManifest => currentManifestIds.Contains(rollbackManifest.Id));
+                }
+
+                fullSet.AddRange(rollbacks);
+            }
+
+            int size = fullSet.Count;
+            fullSet = fullSet.DistinctBy<(ManifestId, ManifestVersionWithBand), ManifestId>(update => update.Item1).ToList();
+            if (size != fullSet.Count)
+            {
+                throw new Exception();
```

Comment:
We shouldn't throw an exception with no details here.  It looks like this is probably supposed to catch the case where the same manifest ID appears in multiple json files in the workload set package.  If so, ideally the message would include information about which ID was duplicated.  However, since in practice we don't expect multiple json files in the package, it's not a big deal.  But we still shouldn't throw an Exception without a message.

Reply:
Good catch; that was supposed to have a real exception, but then I forgot to add that part 🙂